Add services to remote unlock and override pet location

Alex-ala opened the pull request:
Working on top of the device_tracker implementation I've added a service to override the current location if detection was off or if the cat entered/left the house in other ways. This way one can keep the tracker up to date or update it by other external sensors.

<img width="2117" height="684" alt="image" src="https://github.com/user-attachments/assets/5fdf5492-eac1-4503-a96c-cc134f785564" />

In addition to that, another service was added to remote unlock the flap temporarily just like via app:

<img width="2101" height="542" alt="image" src="https://github.com/user-attachments/assets/9e4ed699-e949-4804-a5cb-1515e0724520" />

Im not totally sure if the approach using a `device_action` for the unlock service is correct, but this sounded like the closest match to what this service does.
Also the way to access the correct devices' API client feels weird, but I don't know any other way.

heisenberg2980 commented:
> To get the tests for HACS validation running, can you please add a description and topic tags to this repository @heisenberg2980 ? This would one day allow this repository and the integration being listed in HACS by default. https://github.com/OnlyCatAI/onlycat-home-assistant/actions/runs/16192366437/job/45710875481#step:3:11

Done. Let me know when this PR is ready to be merged, and great job guys!

markusdlugi commented:
@Alex-ala, nice job with the new services. I tested them and can confirm that it's working as intended.

I have two points of feedback:

1. Currently, performing a remote unlock is not reflected in the lock sensor, which could confuse users that might expect this. Maybe extending the [is_unlocked_by_event](https://github.com/OnlyCatAI/onlycat-home-assistant/blob/master/custom_components/onlycat/data/device.py#L102) method to always return true in case of a REMOTE_UNLOCK event trigger source is already sufficient to accomplish this.
2. I believe the remote unlock functionality would be even better integrated as a [button entity](https://developers.home-assistant.io/docs/core/entity/button/). I know that one from the [Shelly integration](https://github.com/home-assistant/core/blob/dev/homeassistant/components/shelly/button.py), in case you need a reference. By using a button, it would be listed separately in the "Configuration" section of the OnlyCat device and would therefore be more accessible than a service which is pretty hidden. Additionally, you wouldn't have to jump through so many hoops to make the API call 😄  But if you want, we can also go with the service for now and add a button later on, would also be fine for me.

While we're at the topic of HACS validation: I recently [created a PR](https://github.com/home-assistant/brands/pull/7426) to get the OnlyCat icon added to the brands repository. It's still waiting for review from a maintainer - but once that has been merged we could enable the brands validation in the Validate workflow. Then we could try and get the repo added as [default repository in HACS](https://hacs.xyz/docs/publish/include/) so that users wouldn't need to manually add it as custom repository anymore.

Alex-ala commented:
> Btw, I was just wondering, how did you learn about the `runDeviceCommand` API call to perform the remote unlock? It's not included in the reference implementation provided by OnlyCat. If there's more commands, e.g. a command for rebooting, those might also be good candidates for the integration.

I've got a MITMproxy setup ready that I also used to understand the websocket communication in the first place. I was completely unfamiliar with websockets and looking at actual traffic helps massively.

I used this setup to record the traffic happening when I use the remote unlock via app. Same could be done for any other action in the app. Currently I'm only aware of the reboot function that is not documented, but if you have any other, I can send a traffic dump.

Alex-ala commented:
Implemented feedback:
1. Turned Unlock service into a button entity
2. The binary sensor for lock state will now be unlocked on trigger_source REMOTE, disregarding the policy
While doing so I also did:
3. Add reboot as a button as well
4. Fixed an edge case error when the integration did not recieve the first event including the trigger_source by making UNKONWN the default trigger_source
5. Fixed an error that the device_id is not within the messages body when a device is reported disconnected by the api.

markusdlugi commented:
Good job! I tested it and like the new implementation as buttons, working as expected of course. The pet location setting service is also useful - I included that in a small automation so I can also manually override it with a button, but I could also imagine to combine it with another motion sensor of some sort, so it's really flexible.

Minor nitpick: You could also adjust the readme to advertise the new buttons 😄
